Build client WASM fails with Rust `1.85`

## Why
The build of the WASM client fails in the CI since the release of Rust `1.85`. 

:warning: This currently blocks the delivery pipeline.

The error is:
```
[INFO]: ⬇️  Installing wasm-bindgen...
thread 'main' panicked at crates/wasm-interpreter/src/lib.rs:245:21:
mithril_common::signable_builder::interface::_::__ctor::h4977fb9f7c35308c: Read a negative address value from the stack. Did we run out of memory?
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Error: Running the wasm-bindgen CLI
Caused by: Running the wasm-bindgen CLI
Caused by: failed to execute `wasm-bindgen`: exited with exit status: 101
  full command: "/home/jp/.cache/.wasm-pack/wasm-bindgen-c59d5019a2b42393/wasm-bindgen" "/home/jp/Works/Cardano/Mithril/mithril/target/wasm32-unknown-unknown/release/mithril_client_wasm.wasm" "--out-dir" "/home/jp/Works/Cardano/Mithril/mithril/mithril-client-wasm/dist/node" "--typescript" "--target" "nodejs"
make: *** [Makefile:13: build] Error 1
```
See https://github.com/input-output-hk/mithril/actions/runs/13441903928/job/37559631936#step:5:203 for more details

## What
Investigate and fix the problem

## How

- [x] Investigate the problem
- [ ] Create a fix

--- a/mithril-common/Cargo.toml
+++ b/mithril-common/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "mithril-common"
-version = "0.5.4"
+version = "0.5.5"
 description = "Common types, interfaces, and utilities for Mithril nodes."
 authors = { workspace = true }
 edition = { workspace = true }

--- a/mithril-common/src/signable_builder/interface.rs
+++ b/mithril-common/src/signable_builder/interface.rs
@@ -17,7 +17,7 @@ use mockall::automock;
 pub trait Beacon: Send + Sync {}
 
 /// Artifact is a trait for types that represent signed artifacts
-#[typetag::serde(tag = "type")]
+#[cfg_attr(not(target_family = "wasm"), typetag::serde(tag = "type"))]
 pub trait Artifact: Debug + Send + Sync {
     /// Get artifact identifier
     fn get_id(&self) -> String;
@@ -54,35 +54,35 @@ impl Beacon for CardanoDbBeacon {}
 
 impl Beacon for Epoch {}
 
-#[typetag::serde]
+#[cfg_attr(not(target_family = "wasm"), typetag::serde)]
 impl Artifact for CardanoDatabaseSnapshot {
     fn get_id(&self) -> String {
         self.hash.clone()
     }
 }
 
-#[typetag::serde]
+#[cfg_attr(not(target_family = "wasm"), typetag::serde)]
 impl Artifact for CardanoStakeDistribution {
     fn get_id(&self) -> String {
         self.hash.clone()
     }
 }
 
-#[typetag::serde]
+#[cfg_attr(not(target_family = "wasm"), typetag::serde)]
 impl Artifact for CardanoTransactionsSnapshot {
     fn get_id(&self) -> String {
         self.hash.clone()
     }
 }
 
-#[typetag::serde]
+#[cfg_attr(not(target_family = "wasm"), typetag::serde)]
 impl Artifact for MithrilStakeDistribution {
     fn get_id(&self) -> String {
         self.hash.clone()
     }
 }
 
-#[typetag::serde]
+#[cfg_attr(not(target_family = "wasm"), typetag::serde)]
 impl Artifact for Snapshot {
     fn get_id(&self) -> String {
         self.digest.clone()

